src/halmos/__main__.py

```diff
@@ -467,22 +464,20 @@ def run(
     else:
         models = [gen_model(args, idx, ex) for idx, ex in execs_to_model]
 
-    end = timer()
+    timer.stop()
 
     no_counterexample = all(m.model is None for m in models)
     passed = no_counterexample and normal > 0 and len(stuck) == 0
     if args.error_unknown:
         passed = passed and all(m.result == unsat for m in models)
     passfail = color_good("[PASS]") if passed else color_warn("[FAIL]")
 
-    time_total, time_paths, time_models = end - start, mid - start, end - mid
-    time_info = f"{time_total:0.2f}s"
-    if args.statistics:
-        time_info += f" (paths: {time_paths:0.2f}s, models: {time_models:0.2f}s)"
+    timer.stop()
```

Comment:
oops looks like I got a spurious stop() here

Reply:
when you stop() here, the end time of the previous subtimers ("paths" in this case) are reset again, which results in a wrong elapsed time for them.